### PR TITLE
Refactor code pull request

### PR DESCRIPTION
**Pull request by AleWaweru**
- [ ] Refactored the main.rb file 
- [ ] Followed the  SOLID principles

### CODE REVIEW

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Alex & Toyo, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🚦 __🏃 🏡 🏃‍♂️  🏚️ 🌴

## Highlights 👍  
- Good job so far working on clean code principles!
- ReadMe is near professional!

## Required Changes ♻️
- [ ] Good job so far, however, we can further improve upon our _create_ and _list_rental_ method in the `# app.rb`, which at present do not adhere to the single responsibility and the YAGNI rule. Let's ensure that each _create_, or _list_ method simply which are discrete on doing a single/particular task and should contain only objects or instances needed for a given logic of that method. 

  Other logics such as requests for input _(`get.chomps`)_ or _input validations\print statements_ should be placed in the `main.rb`, separate classes or methods. Let's therefore refactor the below methods in the `app.rb` accordingly:
  * `create_student` 
  * `create_teacher`
  * `create_book`,
  * `create_rental` and
  * `list_all_rentals` 

  The example below are methods that have focused responsibility:
```ruby
    class App
  
     . . .
  
      def list_books
        @books.each do |book|
          puts book
        end
      end

      def list_people
        @people.each do |person|
          puts person
       end

      . . .

    end
```

I urge you to read further on the same via [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
Generally, it's good idea the subheading text _Author1_ and _Author2_ or similar are replaced with your respective full names in the ReadMe.
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 1]
```diff
@@ -44,14 +44,8 @@ def create_student
...
-      student = Student.new(age, 'classroom', name: name, parent_permission: false)
-      @people << student
-    when 'Y'
-      student = Student.new(age, 'classroom', name: name, parent_permission: true)
-      @people << student
-    end
+    student = Student.new(age, 'classroom', name: name, parent_permission: permission == 'Y')
+    @people << student
     puts 'Student created successfully'
   end
```
Comment (Gambit142) [inline app.rb]: - [ ] I would like to reiterate what the previous reviewer stated. All codes responsible for receiving inputs or codes responsible for validating the input from the user should be put in the main.rb file or a separate class or class method or module. The other methods for creating a class instance or listing instances should only create those instances. 👍 😎 

---

For Example, your `create_book` method should only create a new instance of the `Book` class while codes in the method for getting the `title` and `author` of the _**book**_  from the console app should be put in a separate method, class or module. In the same vein, in the `create_teacher` method, all codes that are responsible for getting the teacher's `name`, `age` and `specialization` should be put in a separate class, module or method while the `create_teacher` method should be responsible for just creating a new instance of the `Teacher` class. 💻 

--- 

Kindly apply these changes for the following methods of the `App` class in the `app.rb` file:

- `create_person`
- `create_student`
- `create_teacher`
- `create_book`
- `create_rental`
- `list_all_rentals`

---

#### N.B: Please ensure that the app works as it should after refactoring the code. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @AleWaweru @toyman640 👋, 

Great work on making the changes requested by a previous reviewer 👏 

## Highlights
- Setup linters correctly ✔️ 
- README file is professional ✔️

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (AleWaweru) [reply app.rb]: done.

**Review by danmainah — APPROVED** [review]
Hi @AleWaweru and @toyman640 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change requested to ensure the Single Responsibility rule is being followed 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**